bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -252,6 +252,48 @@ impl Connection {
         Ok(self)
     }
 
+    /// Signals the connection to do a key_update at the next possible opportunity.
+    /// Note that the resulting key update message will not be sent until `send` is
+    /// called on the connection.
+    ///
+    /// `peer_request` indicates if a key update should also be requested
+    /// of the peer. When set to `KeyUpdateNotRequested`, then only the sending
+    /// key of the connection will be updated. If set to `KeyUpdateRequested`, then
+    /// the sending key of conn will be updated AND the peer will be requested to
+    /// update their sending key. Note that s2n-tls currently only supports
+    /// `peer_request` being set to `KeyUpdateNotRequested` and will return an error
+    /// if any other value is used.
+    pub fn request_key_update(&mut self, peer_request: PeerKeyUpdate) -> Result<&mut Self, Error> {
```

Comment:
Should this return &mut Self, or just ()? I can't imagine this is going to be chained with other calls.

Reply:
@camshaft Do you have an opinion on this? I had initially just copied an adjacent method and forgotten to change the return type. I agree with @lrstewart that `Result<(), Error>` makes more sense for the return type.

But it looks like nearly all of our methods return `&mut Self`, even when they aren't "builder type" methods. https://github.com/aws/s2n-tls/blob/ba825b878a58079e7453fe3c6f00b48a8752398c/bindings/rust/s2n-tls/src/connection.rs#L407

Reply:
I would say consistency is probably the way to go here. Returning `&mut Self` should be a no-op.